src/test/java/io/github/jbellis/brokk/context/ContextSerializationTest.java

```diff
@@ -560,27 +561,36 @@ void testSaveAndLoadSessionHistory() throws Exception {
         
         // Save history
         sessionManager.saveHistory(originalHistory, sessionId);
-        
+
+        // --- Verification with live session manager (cached sessions) ---
+        verifySessionHistory(sessionManager, sessionId, initialModifiedTime, originalHistory, "after save (cached sessions)");
+        project.close();
+
+        // --- Verification with new session manager (sessions loaded from disk) ---
+        MainProject newProject = new MainProject(tempDir);
+        verifySessionHistory(newProject.getSessionManager(), sessionId, initialModifiedTime, originalHistory, "after recreating project (sessions loaded from disk)");
+        newProject.close();
```

Comment:
Extended the test case now to also reload and verify all session data from disk from a new project, because  everything worked fine before from cached session info.